packages/components/src/ui/create-styles/components/theme-provider/theme-provider-context.js

```diff
@@ -0,0 +1,90 @@
+/**
+ * WordPress dependencies
+ */
+import { createContext, useContext, useRef } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import {
+	COLOR_BLIND_MODE_ATTR_PROP,
+	DARK_MODE_ATTR_PROP,
+	HIGH_CONTRAST_MODE_ATTR_PROP,
+	REDUCED_MOTION_MODE_ATTR_PROP,
+} from '../../create-style-system/constants';
+
+/**
+ * @typedef ThemeProviderContext
+ * @property {boolean | null} [isDark] Whether dark mode is enabled.
+ * @property {boolean | null} [isColorBlind] Whether color blind mode is enabled.
+ * @property {boolean | null} [isReducedMotion] Whether reduced motion is enabled.
+ * @property {boolean | null} [isHighContrast] Whether high contrast is enabled.
+ */
+
+export const ThemeProviderContext = createContext(
+	/** @type {ThemeProviderContext} */ ( {
+		isDark: null,
+		isColorBlind: null,
+		isReducedMotion: null,
```

Comment:
We have our own way of checking whether we're in reduced motion mode, I feel like we should remove this from there and rely on the existing hook instead (also in compose package). In other words, why is this a "theme" property?